pkg/maps/ctmap/ctmap.go

```diff
@@ -322,17 +321,20 @@ func doGC4(m *bpf.Map, filter *GCFilter) int {
 
 			del = true
 		}
-		n := nextKey.ToHost().(*CtKey4Global)
+
 		if filter.fType&GCFilterByID != 0 &&
 			// In CT's entries, saddr is the packet's receiver,
 			// which means, is the destination container IP.
-			n.saddr.IP().Equal(filter.IP) {
+			nextKey.saddr.IP().Equal(filter.IP) {
 
 			ruleCtx := policy.RuleContext{
-				SecID: policy.NumericIdentity(entry.src_sec_id),
-				Port:  n.sport,
-				Proto: uint8(n.nexthdr),
+				SecID:          policy.NumericIdentity(entry.src_sec_id),
+				Port:           nextKey.sport,
+				Proto:          uint8(nextKey.nexthdr),
+				L7RedirectPort: entry.proxy_port,
```

Comment:
Same proxy_port usage as comment above. If the above needs to change, so does this.